src/components/smart_objects/src/object_schema_item.cc

```diff
@@ -251,25 +252,23 @@ CObjectSchemaItem::CObjectSchemaItem(const Members& members)
 
 void CObjectSchemaItem::RemoveFakeParams(
     SmartObject& Object, const utils::SemanticVersion& MessageVersion) {
-  for (SmartMap::const_iterator it = Object.map_begin();
-       it != Object.map_end();) {
+  for (SmartMap::const_iterator it = Object.map_begin(); it != Object.map_end();
+       ++it) {
     const std::string& key = it->first;
     std::map<std::string, SMember>::const_iterator members_it =
         mMembers.find(key);
-    if (mMembers.end() == members_it
-        // FIXME(EZamakhov): Remove illegal usage of filed in AM
-        && key.compare(connection_key) != 0 && key.compare(binary_data) != 0 &&
-        key.compare(app_id) != 0) {
-      ++it;
-      Object.erase(key);
 
-    } else if (mMembers.end() != members_it &&
-               GetCorrectMember(members_it->second, MessageVersion)
-                   .mIsRemoved) {
-      ++it;
+    if (mMembers.end() == members_it && key.compare(connection_key) != 0 &&
+        key.compare(binary_data) != 0 && key.compare(app_id) != 0) {
       Object.erase(key);
-    } else {
-      ++it;
+    }
+
+    if (mMembers.end() != members_it) {
```

Comment:
@YarikMamykin how about to swap comparisons with the previous one and use `continue;`?

Reply:
@mkorniichuk , fixed in 29d5f70ba69b90cc1f4bfcb30cdc4fbe7028e7a4